lobsterpy/featurize/core.py

```diff
@@ -873,7 +1119,11 @@ def _calc_ionicity(self) -> float:
                 )
             ):
                 valence_elec = element(structure.species[i].value)
-                val = j / (valence_elec.nvalence() - 0)
+                val = (
+                    j / (valence_elec.nvalence() - 0)
+                    if valence_elec.nvalence() != tol
```

Comment:
Could you check this here as well? I am also not sure what nvalence is and whether != is really a good way to compare floats or integers.

Reply:
it is integer number always. I want to avoid zero divison error here

Reply:
But what is "tol"? This sounds like a float

Reply:
Also line before: why -0?

Reply:
Sorry about that dumb mistake, -0 does not make sense. I fixed it now using explicit try except block